### ISSUE TITLE
TypeError: *parent*.__init__() must be called when overriding __init__

### ISSUE THREAD

**Issue by HeavensInc** [glotzerlab/hoomd-blue#826]
## Description

Python requires a derived class to construct every parent class. So far I found two places where python complains. There might be more though.

## Script

```python
import hoomd

cpu = hoomd.device.CPU()
sim = hoomd.Simulation(device=cpu)
```

## Output

```
Traceback (most recent call last):
  File "test_octahedron_example.py", line 5, in <module>
    import hoomd
  File "[...] hoomd-blue-3beta1/build/hoomd/__init__.py", line 24, in <module>
    from hoomd import write
  File "[...] hoomd-blue-3beta1/build/hoomd/write/__init__.py", line 2, in <module>
    from hoomd.write.gsd import GSD
  File "[...] hoomd-blue-3beta1/build/hoomd/write/gsd.py", line 17, in <module>
    class GSD(Writer):
  File "[...] hoomd-blue-3beta1/build/hoomd/write/gsd.py", line 128, in GSD
    filter=All(),
TypeError: hoomd._hoomd.ParticleFilter.__init__() must be called when overriding __init__
```

## Expected output

None

## Configuration

Platform:
- Linux
- CPU

Installation method:
- Compiled from source

Versions:
- Python version: 3.6.9
- **HOOMD-blue** version: v3.0.0-beta.1-30-g5e466c101 (plus https://github.com/glotzerlab/hoomd-blue/commit/0e5515ef3510590a2f0b8d90f23798575d5eb5c6 )

## Developer

https://github.com/glotzerlab/hoomd-blue/blob/5e466c101ad82fdfe00817a3710954fd902bc184/hoomd/filter/all_.py#L13-L14
```python
class All(ParticleFilter, ParticleFilterAll):
    """Select all particles in the system.

    Base: `ParticleFilter`
    """

    def __init__(self):
        ParticleFilter.__init__(self)
        ParticleFilterAll.__init__(self)
```

This applies to the periodic trigger as well
https://github.com/glotzerlab/hoomd-blue/blob/5e466c101ad82fdfe00817a3710954fd902bc184/hoomd/trigger.py#L89-L90
```python
class Periodic(_hoomd.PeriodicTrigger, Trigger):
[...]
    def __init__(self, period, phase=0):
        Trigger.__init__(self)
        _hoomd.PeriodicTrigger.__init__(self, period, phase)
```

**Comment by joaander** [comment]
@b-butler Why don't we see these errors in our unit tests? We test on python 3.6.

**Comment by joaander** [comment]
Nevermind, now we are getting the failure. The problem is only triggered with the just released **pybind 2.6**.
https://dev.azure.com/glotzerlab/hoomd-blue/_build/results?buildId=6700&view=logs&j=7300be9e-b968-5142-d980-c4de163f8aef&t=476a7dad-4279-5912-0129-66e73a6373d9&l=52
